cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -1997,3 +2015,64 @@
         (value_regs
           (alu_rrr op ty x_lo y_lo)
           (alu_rrr op ty x_hi y_hi))))
+
+;; Float vector compare helpers ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(decl zero_value(Imm64) Imm64)
+(extern extractor zero_value zero_value)
+
+(decl zero_value_f32(Ieee32) Ieee32)
+(extern extractor zero_value_f32 zero_value_f32)
+
+(decl zero_value_f64(Ieee64) Ieee64)
+(extern extractor zero_value_f64 zero_value_f64)
+
+(decl fcompare (FloatCC) VecMisc2)
+(extern constructor fcompare fcompare)
+
+(decl fcompare_swap (FloatCC) VecMisc2)
+(extern constructor fcompare_swap fcompare_swap)
+
+(decl fcmp_cond_zero (FloatCC) FloatCC)
+(extern extractor fcmp_cond_zero fcmp_cond_zero)
+
+(decl fcmp_cond_notEq (FloatCC) FloatCC)
+(extern extractor fcmp_cond_notEq fcmp_cond_notEq)
+
+(decl f_comp_inst (FloatCC Reg VectorSize) Reg)
+(rule (f_comp_inst cond rn size)
+      (vec_misc (fcompare cond) rn size))
+
+(decl f_comp_inst_swap (FloatCC Reg VectorSize) Reg)
+(rule (f_comp_inst_swap cond rn size)
+      (vec_misc (fcompare_swap cond) rn size))
+
+(decl fcmeq0 (Reg VectorSize) Reg)
+(rule (fcmeq0 rn size)
+      (vec_misc (VecMisc2.Fcmeq0) rn size))
+
+;; Int vector compare helpers ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(decl icompare (IntCC) VecMisc2)
+(extern constructor icompare icompare)
+
+(decl icompare_swap (IntCC) VecMisc2)
+(extern constructor icompare_swap icompare_swap)
+
+(decl icmp_cond_zero (IntCC) IntCC)
+(extern extractor icmp_cond_zero icmp_cond_zero)
+
+(decl icmp_cond_notEq (IntCC) IntCC)
+(extern extractor icmp_cond_notEq icmp_cond_notEq)
```

Comment:
Tiny nit but we've mostly stuck to all-lowercase with underscores ("snake case"); would prefer `noteq` or `not_eq` here.